pkgs/stdenv/default.nix

```diff
@@ -39,6 +39,12 @@ rec {
   # Darwin standard environment.
   stdenvDarwin = (import ./darwin { inherit system allPackages platform config;}).stdenvDarwin;
 
+  # Pure Darwin standard environment. Allows building with the sandbox enabled. To use,
+  # you can add this to your nixpkgs config:
+  #
+  #   replaceStdenv = {pkgs}: pkgs.allStdenvs.stdenvDarwinPure
+  stdenvDarwinPure = (import ./pure-darwin { inherit system allPackages platform config;}).stage5;
```

Comment:
`stdenvDarwinPure` is introduced here, but is returned below. This explains the complaint about `/bin/sh` not being available when the sandbox is enabled; that error is coming from the old `darwin/default.nix`.

/cc @pikajude @copumpkin